pandora-client-web/src/networking/socketio_directory_connector.ts

```diff
@@ -383,4 +384,28 @@ export class SocketIODirectoryConnector extends ConnectionBase<IClientDirectory,
 		this.sendMessage('disconnectCharacter', EMPTY);
 		this._lastSelectedCharacter.value = undefined;
 	}
+
+	public async extendAuthToken(password: string): Promise<boolean> {
+		const username = this._authToken.value?.username;
+		if (!username) {
+			throw new Error('Not logged in');
+		}
+		const passwordSha512 = await PrehashPassword(password);
+		const response = await this.awaitResponse('extendLoginToken', { passwordSha512 });
+		switch (response.result) {
+			case 'ok':
+				this._authToken.value = {
+					username,
+					value: response.token,
+					expires: response.expires,
+				};
+				toast('Session extended', TOAST_OPTIONS_SUCCESS);
```

Comment:
Nit: Why is the directory connector responsible for showing the toast with the result? I think it is more sensible to keep UI responsibility to the caller. The directory connector has awfully many responsibilities as it is and they should be more moved out of it instead of new ones being added.